rfcs/202250708-removal-of-graphs/README.md

```diff
@@ -0,0 +1,57 @@
+# Removal of Graph functionalities from oneDAL
+
+## Introduction
+
+OneDAL contains some data structures and algorithms for dealing with graphs:
+https://github.com/uxlfoundation/oneDAL/tree/main/cpp/oneapi/dal/graph
+
+These include, among others:
+* A data structure for representation of graphs.
+* A few standard graph routines such as finding shortest paths, connected
+  components, and counting triangles.
+* Jaccard similarity computations.
+* A clustering algorithm based on Louvain modularity.
+
+However, these functionalities are very underdeveloped - for example, they do
+not contain the same kinds of optimizations used elsewhere in the library (like
+compiler pragmas, intrinsics, blocking optimized for certain cache sizes, calls
+to MKL functions that aren't in NetLib BLAS, etc.), and have no SYCL-based
+implementations to run on GPUs or in SPMD mode.
+
+These graph algorithms have been in preview mode for a while, and have never
+been exposed from the Python side in daal4py, nor in sklearnex - should be noted
+that scikit-learn itself does not offer this type of functionalities, so there
+is no analog to patch from the sklearnex side either.
+
+In their current state, it's unclear whether they are helpful at all. No
+profiling experiments have been conducted so far, but given that they do not
+employ the same kind of optimization tricks seen elsewhere in the library, it's
+unlikely that we would find their performance to be competitive against other
+optimized graph libraries for CPU. And in the case of GPU, there is the
+SyGraph library on which we might focus our efforts on in the future instead:
+https://github.com/unisa-hpc/SYgraph
+
+None of the current team members from Intel side have been involved in the
```

Comment:
I would drop first sentence - first there are people who were involved. 

There are no plans to continue with existing C++ path

I would argue on demand - as we see evolution of NetworkX, cugraph, graphBLAS, SuiteSparse